src/wasm/wasm.cpp

```diff
@@ -1008,7 +1008,18 @@ void CallRef::finalize() {
     return;
   }
   assert(target->type.isRef());
-  if (target->type.getHeapType().isBottom()) {
+  if (target->type.isNull()) {
+    // See StructRef for explanation.
+    if (type.isRef()) {
```

Comment:
Also, StructGet is below - maybe put the comment here and refer to it from below?